tests/integration/conftest.py

```diff
@@ -107,7 +107,7 @@ async def synapse_app_fixture(
 ):
     """Build and deploy the Synapse charm."""
     use_existing = pytestconfig.getoption("--use-existing", default=False)
-    if use_existing:
+    if use_existing or synapse_app_name in model.applications:
```

Comment:
IINM, the ops_test fixture is module scoped, and this fixture is module scoped, so it should not be called twice for the same model (unless you run several modules with --keep-models and --model). In any case, it can be at least helpful for that case, so I close the comment.

Reply:
I agree with you, I thought the same way. But I have no other explanation of why this happens.